src/hooks/usePagination/useQueryString.ts

```diff
@@ -0,0 +1,75 @@
+import React, { useMemo } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import { QueryStringParams } from 'types/recipes';
+
+export const DEFAULT_PAGE = '1';
+export const DEFAULT_PER_PAGE = '10';
+const PER_PAGE_PARAM = 'per_page';
+const PAGE_PARAM = 'page';
+const supportedParams = [PAGE_PARAM, PER_PAGE_PARAM];
+
+export const useQueryString = () => {
```

Comment:
```suggestion
export const useQueryString = <T>() => {
```

Let's define a typescript generic type to tell TS what we expect from query string.